Cargo.toml

```diff
@@ -12,25 +12,25 @@ readme = "README.md"
 edition = "2018"
 
 [dependencies]
-base64 = "0.12"
+base64 = "0.13"
 chrono = { version = "0.4", features = ["serde"] }
 futures = "0.3"
 derive_more = "0.99"
-hmac = "0.8"
-hyper = "0.13"
-hyper-rustls = { version = "0.20", optional = true, default-features = false }
-hyper-tls = { version = "0.4", optional = true }
+hmac = "0.10"
+hyper = { version = "0.14", features = ["http1", "http2", "client", "stream"] }
```

Comment:
In addition to enabling `hyper`'s `http2` feature, it is also necessary to configure ALPN for the underlying TLS connector to enable HTTP/2 (see [h2/akamai.rs](https://github.com/hyperium/h2/blob/e7e75bf1171d33ca2cf7ce34733ee86f4f9b7ae5/examples/akamai.rs) for example). `native-tls` has recently added support for ALPN, so the configuration is probably easy, but that might be out of scope for this PR.

Reply:
Seems easy to do given how it was necessary anyways to make completely separate functions for each of the connector types.

Reply:
Okay, I looked into it now and ALPN support seems to not be a thing of `egg_mode` since we're using `hyper` and not `h2`.

Two relevant crates we use play a role:
`hyper-rustls` actually already supports ALPN: https://github.com/ctz/hyper-rustls/blob/418b7ba167efc7c95dbb154a476c2e589fc367cf/src/connector.rs#L65
`hyper-tls` has an ongoing issue that might actually be solved some time soon: https://github.com/hyperium/hyper-tls/issues/61 Then from what I understand ALPN support will come to egg_mode by simply updating `hyper-tls` once that issue is solved.